vault/resource_pki_secret_backend_config_est.go

```diff
@@ -0,0 +1,181 @@
+				Description: "A pairing of EST label to the configured EST behavior for it",
+			},
+			consts.FieldAuthenticators: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Description: "Lists the mount accessors EST should delegate authentication requests towards",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+						"userpass": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+			consts.FieldEnableSentinelParsing: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Enable parsing of fields from the provided CSR for Sentinel policies",
+			},
+			consts.FieldAuditFields: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Fields parsed from the CSR that appear in the audit and can be used by sentinel policies",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func pkiSecretBackendConfigEstWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := verifyPkiEstFeatureSupported(meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	path := pkiSecretBackendConfigEstPath(backend)
+
+	authenticators := d.Get(consts.FieldAuthenticators).([]interface{})
+	var authenticator interface{}
+	if len(authenticators) > 0 {
+		authenticator = authenticators[0]
+	}
+
+	data := map[string]interface{}{
+		consts.FieldEnabled:               d.Get(consts.FieldEnabled).(bool),
+		consts.FieldDefaultMount:          d.Get(consts.FieldDefaultMount).(bool),
+		consts.FieldDefaultPathPolicy:     d.Get(consts.FieldDefaultPathPolicy).(string),
+		consts.FieldLabelToPathPolicy:     d.Get(consts.FieldLabelToPathPolicy).(map[string]interface{}),
+		consts.FieldAuthenticators:        authenticator,
+		consts.FieldEnableSentinelParsing: d.Get(consts.FieldEnableSentinelParsing).(bool),
+		consts.FieldAuditFields:           d.Get(consts.FieldAuditFields).([]interface{}),
+	}
+
+	log.Printf("[DEBUG] Updating EST config on PKI secret backend %q:\n%v", backend, data)
+	_, err := client.Logical().WriteWithContext(ctx, path, data)
+	if err != nil {
+		return diag.Errorf("error updating EST config for PKI secret backend %q: %s", backend, err)
+	}
+	log.Printf("[DEBUG] Updated EST config on PKI secret backend %q", backend)
+
+	d.SetId(path)
+
+	return pkiSecretBackendConfigEstRead(ctx, d, meta)
+}
+
+func pkiSecretBackendConfigEstRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return readPKISecretBackendConfigEst(ctx, d, meta)
+}
+
+func pkiSecretBackendConfigEstDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Looks like there is no `DELETE` method on this endpoint (makes sense, config endpoints in Vault usually don't). In that case we can just `return nil` for this method, akin to what is done for the [KVV2 Config resource](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kv_secret_backend_v2.go#L135-L137)

We let users control the updates on the configs (which includes resets, a form of update in this case) by updating their TF config file. This keeps it in line w/ how a user would expect to perform the reset in Vault (a `vault write pki/config/est` as opposed to `vault delete pki/config/est`).

This also enables the user to keep track of their reset EST config in the TF state. This method is called upon a `terraform destroy`, so TF would also nuke the state of the EST config resource when it gets reset (unless there's an actual removal of the resource in Vault, TF practice would be to manage the lifecycle for as long as possible 😄 ).

Reply:
Fixed in https://github.com/hashicorp/terraform-provider-vault/pull/2246/commits/8183fe41f06467d6c25bb1f4b6484bbfeff8bf2e